### PR TITLE
[Core] Allow to configure dashboard agent grpc port.

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

Currently, there's no way to specify grpc port for dashboard agents (only http ports are allowed to be configured). This PR allows users to do that.

This also add more features; 
- Format the port conflict error message better. I found the default worker ports are from 10002-19999, which spams the terminal. It is now formatted as 10002-19999.
- Add dashboard http port to the port conflict list. 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/_private/parameter.py [hunk 1]
```diff
@@ -316,7 +318,8 @@ def wrap_port(port):
                         f"Ray component {comp} is trying to use "
                         f"a port number {port} that is used by "
                         "other components.\n"
-                        f"Port information: {pre_selected_ports}\n"
+                        f"Port information: "
+                        f"{self._format_ports(pre_selected_ports)}\n"
```
Comment (iycheng) [inline python/ray/_private/parameter.py:322]: is this related to this PR?

Comment (rkooo567) [reply python/ray/_private/parameter.py:322]: Yes! it is written in the PR description.

**Comment by jjyao** [comment]
Is it related to the dashboard port auto discovery thing that eric and chen worked on before?

**Comment by rkooo567** [comment]
It is not. It is just a way to work around it (that's how OpenAI avoids the issue now)

File: python/ray/_private/parameter.py [hunk 2]
```diff
@@ -392,3 +395,20 @@ def _check_usage(self):
...
+
+        for comp, port_list in ports.items():
+            if len(port_list) == 1:
+                ports[comp] = port_list[0]
+            elif len(port_list) == 0:
+                # Nothing is selected, meaning it will be randomly selected.
+                ports[comp] = "random"
+            elif comp == "worker_ports":
+                ports[comp] = (
+                    f"{port_list[0]}-{port_list[len(port_list) - 1]}")
```
Comment (scv119) [inline python/ray/_private/parameter.py:413]: maybe we can rephrase it a little bit better: i.e. `f"{len(port_list) ports from {min_port} to {max_port}"`

**Review by scv119 — CHANGES_REQUESTED** [review]

**Comment by rkooo567** [comment]
cc @scv119

**Review by scv119 — APPROVED** [review]
